### PR TITLE
Spell book rebalance

### PR DESCRIPTION
**Pull request by ghost**
Spell books now have 10 uses. Most spells cost 2 uses, but some underused spells (like Blind and Forcewall) cost 1.

> Fireball: 2
> Magic Missile: 2
> Disintegrate: 2
> Disable Tech: 1
> Repulse: 2
> Time Stop: 2
> Smoke: 1
> Blind: 1
> Mindswap: 2
> Forcewall: 1
> Blink: 2
> Teleport: 2
> Mutate: 2
> Ethereal Jaunt: 2
> Knock: 1
> Flesh to Stone: 2
> Summon Item: 1
> Bind Soul: 2
> Lightning Bolt: 2
> Barnyard Curse: 2
> Staff of Change, Animation, and Chaos: 2
> Staff of Door Creation: 1
> Scrying Orb: 2
> Soul Shard Belt: 2
> Necromantic Stone: 2
> Wand Assortment: 2
> Mastercrafted Armor Set: 2
> Contract of Apprenticeship: 2
> Bottle of Blood: 2
> Summon Guns: 2
> Summon Magic: **CHALLENGE, REWARDS 1 USE**
> Summon Events: 2

### CODE REVIEW

**Comment by PKPenguin321** [comment]
neat

**Comment by WJohn** [comment]
So when are we giving wizards jaunt and MM to start with? Because utility spells like this have basically no chance of ever getting picked by virtue of them needing two other utility spells and then only barely enough firepower to last a few minutes.

File: code/datums/spells/wizard.dm [hunk 1]
```diff
@@ -333,3 +333,45 @@
...
+		return 0
+	playMagSound()
+	user.visible_message("<span class='warning'>A wreath of flames passes over [user]!</span>", "<span class='userdanger'>You banish your wounds with purifying flame!</span>")
+	user.adjust_fire_stacks(1)
+	user.IgniteMob()
+	user.adjustBruteLoss(-user.getBruteLoss())
+	user.adjustFireLoss(-user.getFireLoss())
+	user.adjustToxLoss(-user.getToxLoss())
+	user.adjustOxyLoss(-user.getOxyLoss())
+	user.adjustCloneLoss(-user.getCloneLoss()) //Too powerful?
```
Comment (Anonus) [inline code/datums/spells/wizard.dm]: It's probably okay.

**Comment by Bawhoppen** [comment]
this being a dumb spell aside, WJ I'd disagree and that thats ok

**Comment by Zilenan91** [comment]
Really though, wizards should probably get MM and jaunt by default nowadays anyways because they're so good and needed every round.

**Comment by Bawhoppen** [comment]
No, the point is a wizard trying to win will always get them, however a FUN wizard might not.

**Comment by WJohn** [comment]
Unfortunately, baw, a "fun" wizard will last only 12 seconds without them.

**Comment by Bawhoppen** [comment]
I guarantee if they're robust they will do better than some powergaming el nath, MM shitter

**Comment by KorPhaeron** [comment]
Good luck ever killing a wizard who gets MM+jaunt for free, because they now have the points for 2 more escape spells that will become the default choices.

**Comment by Xhuis** [comment]
I _suppose_ I could give wizards Magic Missile and Ethereal Jaunt by default in this PR.

**Comment by Bawhoppen** [comment]
Kill yoursel

**Comment by KorPhaeron** [comment]
Finish cult or vampires dont half revamp wizard as well aaaaa

**Comment by Xhuis** [comment]
@KorPhaeron Cult is already finished being revamped, and vampires are on hiatus because I've lost all motivation to work on them.

@Bawhoppen pls

**Comment by Bawhoppen** [comment]
I dont like cancerous changes

**Comment by Xhuis** [comment]
maybe if you actually provided constructive feedback instead of "Kill yoursel" people would be more obligated to listen to you

**Comment by Anonus** [comment]
If you seriously give wizards jaunt and MM I'll just do the perma fire wizard gimmick with this spell and just make absolutely everyone cry because I'll be unkillable and memeing hard.

**Comment by KorPhaeron** [comment]
I think doubling wizard points and then making most skills cost 2 points would be a saner way of making underused spells more attractive (like what we did with uplinks)

**Comment by Xhuis** [comment]
Fair enough, I can definitely see about that

**Comment by Xhuis** [comment]
I did a rebalance of the spell book. It has 10 uses by default now.

**Comment by LanCartwright** [comment]
just fuck my shit up

**Comment by Iamgoofball** [comment]
sign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: Нsign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: Нsign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: Нsign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: Нsign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: Нsign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: Нsign me the FUCK up good shit go౦ԁ sHit thats ✔ some goodshit rightth ere right✔there ✔✔if i do ƽaү so my selｆ :100: i say so :100: thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ:100: Н

**Comment by Lijah99** [comment]
original

**Comment by KorPhaeron** [comment]
How does refunding spells work with variable costs? Was that already taken care of?

**Comment by Xhuis** [comment]
@KorPhaeron It seems like the framework was already implemented, as refunding the spell seems to return all of the points spent on it. I'll do some more testing on this and fix it if it needs to be fixed.

**Comment by Anonus** [comment]
Seems good.
Or I might just want to play a wizard round where I meme the crew to death the worst possible way.

**Comment by Alek2ander** [comment]
Disable tech isn't underused IMO. It's picked pretty often and it's pretty damn useful.

**Comment by KorPhaeron** [comment]
@Alek2ander 

Not quite as great now that steelrifles exist though

**Comment by MrPerson** [comment]
The spell itself is completely useless since it's worse than a simple medkit any wizard can trivially steal.

**Comment by Incoming5643** [comment]
What did the staff of change ever do to you?

File: code/datums/spells/wizard.dm [hunk 2]
```diff
@@ -333,3 +333,45 @@
 				M.Weaken(2)
 				M << "<span class='userdanger'>You're thrown back by a mystical force!</span>"
 			spawn(0) AM.throw_at(throwtarget, ((Clamp((maxthrow - (Clamp(distfromcaster - 2, 0, distfromcaster))), 3, maxthrow))), 1,user)//So stuff gets tossed around at the same time.
+
+/obj/effect/proc_holder/spell/targeted/cauterize
+	name = "Cauterize"
+	desc = "This spell will instantly heal all damage to your body. However, after a short time, half of it will dealt back to you in the form of fire damage."
+	school = "transmutation"
+	charge_max = 300
+	cooldown_min = 200 //25 deciseconds reduction per rank
```
Comment (Incoming5643) [inline code/datums/spells/wizard.dm]: That's not a very charitable scale down, if a wizard wants to go full out x5 Cauterize it should be ridiculously short.

**Comment by Incoming5643** [comment]
What did the staff of change ever do to you? I think both the staff of animation and chaos are far stronger and I know they're taken more often.

**Comment by Incoming5643** [comment]
Spells by current popularity:

<b>Ethereal Jaunt | Currently 2 points | Suggest raise to 3
Magic Missile | Currently 2 points | Suggest raise to 3
Time Stop | Currently 2 points | Suggest raise to 3</b>
Blink | Currently 2 points | Probably fine
Teleport | Currently 2 points | Probably fine
<b>Lightning Bolt | Currently 1 point | STRONGLY suggest raising to 2, it's really good post buff</b>
Fireball | Currently 2 points | Probably fine
Disintegrate | Currently 2 points | Probably fine but someone will probably demand you set it to 3 because they're salty
Bottle of Blood | Currently 2 points | Probably fine because of the 3 limit (not sure why this is as popular as it is)
Mutate | Currently 2 points | Probably fine
Summon Magic | Currently free but one time only | Fine but we really should reunite how summon magic and summon guns work at some point (not this pull)
Summon Guns | Currently 2 points | Probably fine
<i>LichMode</i>™.| Currently 2 points | Probably fine
Staff of Chaos | Currently 2 points | Probably fine
<b>Summon Events | Currently 1 point | STRONGLY suggest raising to 2 (you probably just overlooked this)</b>
Repulse | Currently 2 points | Probably fine
Contract | Currently 2 points | Probably fine
Soul Stone Belt | Currently 2 points | Probably fine
<b>Disable Tech | Currently 1 point | I could go either way between 1 or 2 on this one</b>
Summon Item | Currently 1 point | Probably fine since it's primarily just insurance
<b>Staff of Animation | Currently 2 points | Probably fine but if any of the staffs should be 3 points it's this joker</b>
Smoke | Currently 1 point | Blaze it
<b>Gem Hardsuit | Currently 2 points | Fine but people really under appreciate it, almost want to see it at 1 just so people would give it a chance</b>
<b>Staff of Change | Currently 3 points | STRONGLY suggest droping to 2. Check the wabbajack proc, the odds of getting screwed over by this staff are less than you probably think.</b>
Knock | Currently 1 point | Probably fine
Necromantic Stone | Currently 2 points | Fine but underused and somewhat redundant
Flesh to Stone | Currently 2 points | Fine, if Disintegrate goes to 3 points, this should still be 2
Wand Assortment | Currently 2 points | Probably fine
Mind Transfer | Currently 2 points | Probably fine
<b>Barnyard Curse | Currently 2 points | Almost never used, almost want to see it at 1 just for that reason</b>
Forcewall | Currently 1 point | Probably fine.

Anything I didn't list here <b>hasn't been used once in the last month</b> and needs a lot of love. They should definitely be set to 1. The poor staff of doors could stand to be absolutely free honestly.

**Comment by Alek2ander** [comment]
Scrying Orb is the best thing for any wizard ever and don't let anyone tell you otherwise.

**Comment by CPTANT** [comment]
This is mostly a nerf of the wizard. It will not lead to better gameplay

**Comment by Durkel** [comment]
I wouldn't increase the cost to any spells until after this goes in affect and we see how and if this changes a wizards decision in buying spells. And increasing the cost of the mobility spells is just a bad idea, you might get pissy when he jaunts away but you should realize that it's 1v70 and you've now got steelrifles to use on his ass.

**Comment by Xhuis** [comment]
@CPTANT That makes no sense. There are no effective cost raisings outside of the staff of change, which I've been convinced to lower back down anyway.

**Comment by Xhuis** [comment]
I did some more balancing of spell points. The standard MM and jaunt still cost only 2 points. Also, Cauterize can go up to a 10-second cooldown when fully upgraded.

**Comment by Dorsisdwarf** [comment]
>10 second cooldown for half your spell points and all it does is turn your brute/toxic to burn damage

**Comment by Anonus** [comment]
It halves your damage, though.

**Comment by Xhuis** [comment]
@Dorsisdwarf What Anonus said. This is a foray into giving wizards a healing spell and I didn't want to overdo it by just giving them a static -10 damage or something.

**Comment by Incoming5643** [comment]
@Xhuis Only because they're extremely overused compared to everything else. Literally everyone takes Jaunt and 75% of them will take magic missle. Time stop is currently sitting at 44% so I guess that's a harder sell, but given how immune to counters it is it still feels like a really comically strong spell.

Some people think that Jaunt/Missile are so common that they should be free, but my thought is that having "a standard loadout" for wizard either in code or de facto by spell choice just stifles creativity.

**Comment by Xhuis** [comment]
I don't want to limit spell choice, either. If someone takes Magic Missile and Ethereal Jaunt, they shouldn't only be able to take two more "standard" spells. They're very good spells, but I don't want to discourage use of them.

**Comment by Anonus** [comment]
The gemsuit is really strong, especially if you become a plasman and grab a plasma tank from atmos, engineering, or toxins because then you get free permafire, which is pretty dang strong.
Also, making a spell have 3 or more cost means you won't be able to max its cooldown, which may or may not be a good thing.

**Comment by Miauw62** [comment]
because teleport/ei nath shitwizard is too weak???
wtf is even the point of this spell.

:-1:

**Comment by Xhuis** [comment]
By that logic, we should remove all of the spells at aren't jaunt, MM, ei nath, teleport, and blink. Staff of Change costs 2, I forgot update the OP.

**Comment by Zilenan91** [comment]
It probably was overboard with suggesting MM and jaunt be standard on wizards, so could we possibly just make jaunt standard? Wizard is literally unplayable without it, and if it or MM aren't going to be made standard, they need to stay at 2 (or 1 for jaunt since it's required). They're basically the only things that prevent wizards from getting stuck/dying instantly in the hallways.

Aside from that, I look forward to arming my entire department with fireballs, good PR.

Something else I'd like to see is if the gem hardsuit cost 1 spell point and prevented imbedding of things like spears and stuff, since those are SUCH a meme nowadays and mess up the wizard hard. Hell, a single spear imbedding with no other damage will put you in near-crit, and coupled with bleeding, the wizard is basically dead unless he has brute kits on him or something.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'd much rather see the spell have a chargeup where the more you charge it up the more it heals. As of now, I don't really see anyone using this especially considering after your first 3 kills as lel einathmmjaunt wizard you probably have medbay access to give yourself insta-healing patches. 

Whatever you decide to do with it, please add it to the healing assistant.

**Comment by Xhuis** [comment]
People seemed to not want it, so I scrapped Cauterize for now. I'm still keeping the spellbook rebalance, as I feel some of the spells that get very little use deserve some love.

**Comment by KorPhaeron** [comment]
>people complain for weeks about how dumb fleshmend is
>people complain that a new healing ability isn't over the top enough

Spell book points doubling was a long time coming though, so that's nice

**Comment by Core0verload** [comment]
Please put a list of current spell costs to OP.

**Comment by Xhuis** [comment]
All spell costs have been posted above.

**Comment by Xhuis** [comment]
Any issue with merging this?

**Pull request merged**